eth/client.go

```diff
@@ -246,733 +227,406 @@ func (c *Client) SetManagers() error {
-			glog.Infof("[%v] Submitted tx %v. Deposit %v LPTU", c.account.Address.Hex(), tx.Hash().Hex(), amount)
-			return tx, nil
-		}
-	})
+	return c.BondingManagerSession.WithdrawFees()
 }
 
-func (c *Client) GetBroadcasterDeposit(broadcaster common.Address) (*big.Int, error) {
-	b, err := c.jobsManagerSession.Broadcasters(broadcaster)
+func (c *client) autoClaimTokenPoolsShares() error {
+	dStatus, err := c.DelegatorStatus(c.Account().Address)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return b.Deposit, nil
-}
 
-func (c *Client) WithdrawDeposit() (<-chan types.Receipt, <-chan error) {
-	return c.WaitForReceipt(func() (*types.Transaction, error) {
-		if tx, err := c.jobsManagerSession.Withdraw(); err != nil {
-			return nil, err
-		} else {
-			glog.Infof("[%v] Submitted tx %v. Withdraw deposit", c.account.Address.Hex(), tx.Hash().Hex())
-			return tx, nil
+	if dStatus != 0 {
+		// If already registered, auto claim token pools shares
+		currentRound, err := c.CurrentRound()
+		if err != nil {
+			return err
 		}
-	})
-}
 
-func (c *Client) Job(streamId string, transcodingOptions string, maxPricePerSegment *big.Int, endBlock *big.Int) (<-chan types.Receipt, <-chan error) {
-	return c.WaitForReceipt(func() (*types.Transaction, error) {
-		if tx, err := c.jobsManagerSession.Job(streamId, transcodingOptions, maxPricePerSegment, endBlock); err != nil {
-			return nil, err
-		} else {
-			glog.Infof("[%v] Submitted tx %v. Creating job for stream id %v", c.account.Address.Hex(), tx.Hash().Hex(), streamId)
-			return tx, nil
+		dInfo, err := c.BondingManagerSession.GetDelegator(c.Account().Address)
+		if err != nil {
+			return err
 		}
-	})
-}
 
-func (c *Client) ClaimWork(jobId *big.Int, segmentRange [2]*big.Int, claimRoot [32]byte) (<-chan types.Receipt, <-chan error) {
-	return c.WaitForReceipt(func() (*types.Transaction, error) {
-		if tx, err := c.jobsManagerSession.ClaimWork(jobId, segmentRange, claimRoot); err != nil {
-			return nil, err
-		} else {
-			glog.Infof("[%v] Submitted tx %v. Claimed work for segments %v - %v", c.account.Address.Hex(), tx.Hash().Hex(), segmentRange[0], segmentRange[1])
-			return tx, nil
-		}
-	})
-}
+		lastClaimTokenPoolsSharesRound := dInfo.LastClaimTokenPoolsSharesRound
 
-func (c *Client) Verify(jobId *big.Int, claimId *big.Int, segmentNumber *big.Int, dataStorageHash string, dataHashes [2][32]byte, broadcasterSig []byte, proof []byte) (<-chan types.Receipt, <-chan error) {
-	return c.WaitForReceipt(func() (*types.Transaction, error) {
-		if tx, err := c.jobsManagerSession.Verify(jobId, claimId, segmentNumber, dataStorageHash, dataHashes, broadcasterSig, proof); err != nil {
-			return nil, err
-		} else {
-			glog.Infof("[%v] Submitted tx %v. Verify segment %v in claim %v", c.account.Address.Hex(), tx.Hash().Hex(), segmentNumber, claimId)
-			return tx, nil
-		}
-	})
-}
+		var endRound *big.Int
+		for new(big.Int).Sub(currentRound, lastClaimTokenPoolsSharesRound).Cmp(RoundsPerTokenSharesClaim) == 1 {
```

Comment:
Yeah there could be 0 rewards/fees in a round for a delegator claim - the contract function `claimTokenPoolsShares` skips over that round while looping from the last claim round through the end round if that is the case. We would still need to make a call to `claimTokenPoolsShares` to actually update the delegator's last claim round to the end round though

Reply:
I see... This means if you are a transcoder, you pretty much have to call claimTokenPoolsShares every 20 rounds right?

Reply:
You only need to call claimTokenPoolsShares if you are performing a delegation related action (i.e. Bond, Unbond, WithdrawStake, WithdrawFees). If you're an active transcoder that is just continuously processing jobs you don't need to do anything unless you want to a) bond more tokens to yourself b) withdraw the fees you've collected c) unbond and then withdraw your stake. When you do any of these actions, the client automatically goes through all the rounds since you've last claimed rewards and fees and credits your stake and fees with the appropriate amounts before executing the actual action